src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -633,19 +633,31 @@ function applyLoadoutMods(
     const mods = modHashes.map((h) => defs.InventoryItem.get(h)).filter(isPluggableItem);
 
     // What mods are already on the equipped armor set?
-    const existingMods = new Set<number>();
+    const existingMods: number[] = [];
     for (const item of armor) {
       if (item.sockets) {
         for (const socket of item.sockets.allSockets) {
           if (socket.plugged) {
-            existingMods.add(socket.plugged.plugDef.hash);
+            existingMods.push(socket.plugged.plugDef.hash);
           }
         }
       }
     }
 
     // Early exit - if all the mods are already there, nothing to do
-    if (modHashes.every((h) => existingMods.has(h))) {
+    if (
+      modHashes.every((h) => {
+        const foundAt = existingMods.indexOf(h);
+        if (foundAt === -1) {
+          // a mod was missing
+          return false;
+        } else {
+          // the mod was found, but we have consumed this copy of it
+          delete existingMods[foundAt];
```

Comment:
This'll work but it's a bit weird vs a splice

Reply:
is it though? it's the more obvious action ("delete this"),
splice is many times more complicated/steps internally,
and js array/iteration behaviors automatically ignore unassigned indexes in arrays, so it probably saves indexOf even having to check that index.

Reply:
js users come in expecting to use `delete` then we're trained to use splice because delete does things a newbie might not expect. but there's no surprise/pitfall here

Reply:
Yeah like I said, it totally works fine, it's just a bit weird mostly coming from other languages where the concept of deleting a key out of an array is not a thing.